src/modal.service.ts

```diff
@@ -0,0 +1,30 @@
+import {Injectable} from '@angular/core';
+
+import {ModalComponent} from './modal.component';
+
+@Injectable()
+export class ModalService {
```

Comment:
I'd try to avoid using a service if no config is involved. Whenever a service comes into play so does the [forRoot()](https://github.com/SebastianM/angular-google-maps/blob/master/packages/core/core.module.ts), [to prevent multiple instances](https://stackoverflow.com/a/42866725). I think it doesn't hurt performance much to do a `elementRef.nativeElement.querySelector('input')`

Reply:
Yes, after pushing the commit that includes the service, I realize I got to much cases to handle and din't like my approach. Definitely not a good approach. 

 What's your approach with `elementRef.nativeElement.querySelector('input')`, i don't get it.